### ISSUE TITLE
Rate Limiting

### ISSUE THREAD

**Issue by richraid21** [TylerBrock/saw#11]
It would be really nice if rate limiting was implemented to back off if you use the tool to watch multiple log groups at once. AWS suggests [exponential backup](https://docs.aws.amazon.com/general/latest/gr/api-retries.html). 

This will become very important if #9 starts to be worked on.

**Comment by TylerBrock** [comment]
Yup, agree completely. Thanks for reporting this and for linking to the docs on it. This is definitely on my radar.
